playbooks/roles/mongo_3_2/templates/backup-mongo.sh.j2

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+# ref https://tasks.opencraft.com/browse/SE-1669
+# Script to perform a point-in-time dump of the local mongodb database using
+# mongodump.
+# includes locking (prevent this script from running multiple times in
+# parallel), creating a snapshot of the volume used to backup to, and snitch monitoring
+
+# vars
+ARCHIVE_DIR="TODO"
+ARCHIVE_NAME="mongo-$(date --iso-8601=minutes --utc).gz"
+MONGO_ADMIN_PASSWORD="TODO"
+VOLUME="TODO" # need some way to tell the script which is the volume being used - must be a dedicated ebs
```

Comment:
> need some way to tell the script which is the volume being used

Yep -- this script is a template (`.j2`), so we can use ansible variables to hard-code specific details like this.

Nice that you found the old [backup-mongo-to-s3.j2](https://github.com/edx/configuration/blob/master/playbooks/roles/mongo_2_6/templates/backup-mongo-to-s3.j2) script -- I didn't know that was there!  Can you adapt it here to handle the archive-to-volume logic change that we want to make here?

I've kept the same variable names used by the old script below for clarity, but we could remove the `_S3_` part.  Specifically, you can use ansible variables like:

* `MONGO_S3_BACKUP_NODE` : which mongo node should run the backup.
   Setting something like this lets us deploy this script (and even the cron) to all of our mongo servers, but ensures the backup process is only run on our designated node.
* `MONGO_S3_BACKUP_TEMPDIR`: where the mongodump file goes.
   This is your `ARCHIVE_DIR` above.
* `COMMON_MONGO_READ_ONLY_USER` / `COMMON_MONGO_READ_ONLY_PASS`: credentials used to take the backups.
   They should have sufficient access, so we don't need the full `admin` user?

Then, since we don't want to send the backup to S3, we want to send it to the specific EBS volume, you can gate that with something like:

```
if [[ "{{ MONGO_S3_BACKUP_BUCKET }}"  ]]; then
  # send backup to S3 bucket
  $noop s3cmd -c {{ MONGO_S3_S3CMD_CONFIG }} sync {{ MONGO_S3_BACKUP_TEMPDIR }}/{{ COMMON_ENVIRONMENT }}-{{ COMMON_DEPLOYMENT }}-$DATESTAMP.tar.gz "s3://{{ MONGO_S3_BACKUP_BUCKET }}/mongo/"
elif [[ "{{MONGO_EBS_BACKUP_VOLUME}}" ]]; then
  # mount EBS volume
  # copy new snapshot over
  # ...
fi
```

Reply:
> Can you adapt it here to handle the archive-to-volume logic change that we want to make here?

tbh there isn't much from that script that is worth keeping. Even the mongodump call needs tweaking (doesn't contain --oplog et al) :)

Reply:
I think at the moment I'll implement it as a totally new script, and then we can merge later if required.

Reply:
That's totally fine -- especially since edX haven't carried that script through to later versions of the mongo role.  Though it's useful to observe the conventions used by that old script.